hostinit/setup.go

```diff
@@ -537,22 +522,46 @@ func (init *HostInit) ProvisionHost(ctx context.Context, h *host.Host) error {
 		return errors.Wrapf(err, "error initializing host %s", h.Id)
 	}
 
-	grip.Infof("Setup complete for host %s", h.Id)
-
+	// If this is a spawn host
 	if h.ProvisionOptions != nil &&
-		h.ProvisionOptions.LoadCLI &&
-		h.ProvisionOptions.OwnerId != "" {
+		h.ProvisionOptions.LoadCLI {
+
 		grip.Infof("Uploading client binary to host %s", h.Id)
 		lcr, err := init.LoadClient(ctx, h)
 		if err != nil {
+			err = errors.Wrapf(err, "Failed to load client binary onto host %s: %+v", h.Id, err)
 			grip.Errorf("Failed to load client binary onto host %s: %+v", h.Id, err)
-		} else if err == nil && len(h.ProvisionOptions.TaskId) > 0 {
+			return err
+		}
+
+		if h.ProvisionOptions.OwnerId != "" &&
+			len(h.ProvisionOptions.TaskId) > 0 {
 			grip.Infof("Fetching data for task %s onto host %s", h.ProvisionOptions.TaskId, h.Id)
 			err = init.fetchRemoteTaskData(ctx, h.ProvisionOptions.TaskId, lcr.BinaryPath, lcr.ConfigPath, h)
 			grip.ErrorWhenf(err != nil, "Failed to fetch data onto host %s: %v", h.Id, err)
 		}
+
+		grip.Infof("Running setup script for spawn host %s", h.Id)
+		args := []string{lcr.BinaryPath, "agent", "--setup_only"}
+		if h.Distro.SetupAsSudo {
+			args = append(args, "--setup_as_sudo")
+		}
+		cloudHost, err := providers.GetCloudHost(h, init.Settings)
+		if err != nil {
+			return errors.Wrapf(err, "error getting cloud host for %v", h.Id)
+		}
+		sshOptions, err := cloudHost.GetSSHOptions()
+		if err != nil {
+			return errors.Wrapf(err, "error getting ssh options for host %s", h.Id)
+		}
+		out, err := hostutil.RunRemoteScript(ctx, h, strings.Join(args, " "), sshOptions)
```

Comment:
should we background=true this?

Reply:
No, I think we don't want to continue until it finishes. And it has a timeout.